src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Simplification;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToInterpolatedString
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertToInterpolatedString), Shared]
+    internal partial class ConvertToInterpolatedStringRefactoringProvider : AbstractConvertToInterpolatedStringRefactoringProvider<InterpolatedStringExpressionSyntax, InvocationExpressionSyntax, ExpressionSyntax, ArgumentSyntax, LiteralExpressionSyntax>
+    {
+        protected override SeparatedSyntaxList<ArgumentSyntax>? GetArguments(InvocationExpressionSyntax invocation) =>
+            invocation?.ArgumentList?.Arguments;
+
+        protected override ImmutableArray<ExpressionSyntax> GetExpandedArguments(
+            SemanticModel semanticModel, 
+            SeparatedSyntaxList<ArgumentSyntax> arguments)
+        {
+            var builder = ImmutableArray.CreateBuilder<ExpressionSyntax>();
+            for (int i = 1; i < arguments.Count; i++)
+            {
+                builder.Add(CastAndParenthesize(arguments[i].Expression, semanticModel));
+            }
+
+            var expandedArguments = builder.ToImmutable();
+            return expandedArguments;
+        }
+
+        protected override LiteralExpressionSyntax GetFirstArgument(SeparatedSyntaxList<ArgumentSyntax> arguments) =>
+            arguments[0]?.Expression as LiteralExpressionSyntax;
+
+        protected override InterpolatedStringExpressionSyntax GetInterpolatedString(string text) =>
+            (InterpolatedStringExpressionSyntax)ParseExpression("$" + text);
+
+        protected override string GetText(SeparatedSyntaxList<ArgumentSyntax> arguments)
+        {
+            var stringToken = ((LiteralExpressionSyntax)arguments[0].Expression).Token;
+            var text = stringToken.ToString();
+
+            if (stringToken.IsVerbatimStringLiteral())
+            {
+                // We need to escape braces as this is an ambiguous case in interpolated strings that is 
+                // not ambiguous in verbatim strings
+                text = text.Replace(@"'{'", @"'{{'").Replace(@"'}'", @"'}}'");
+            }
+
+            return text;
+        }
+
+        protected override bool IsArgumentListCorrect(
+            InvocationExpressionSyntax invocation,
+            ISymbol invocationSymbol,
+            ImmutableArray<ISymbol> formatMethods,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            if (invocation.ArgumentList != null &&
```

Comment:
Why not its a class, your saying we should crash if it is?

---
In reply to: [65440236](https://github.com/dotnet/roslyn/pull/11415#discussion_r65440236) [](ancestors = 65440236)